apps/backend/src/services/assetRegistrar/eam/utils/performApiRequest.ts

```diff
@@ -1,32 +1,33 @@
-import { getToken } from '../api/getToken';
 import { createAndLogError } from './createAndLogError';
 import { getEnvOrThrow } from './getEnvOrThrow';
 
-export async function performApiRequest(requestData: string) {
+export async function performApiRequest(uri: string, requestData: object) {
   try {
-    const accessToken = await getToken();
-
-    const response = await fetch(getEnvOrThrow('EAM_API_URL'), {
+    const response = await fetch(getEnvOrThrow('EAM_API_URL') + uri, {
       method: 'POST',
-      body: requestData,
+      body: JSON.stringify(requestData),
       headers: {
-        'Content-Type': 'text/xml',
-        Authorization: `Bearer ${accessToken.token.access_token}`,
+        'Content-Type': 'application/json',
+        INFOR_USER: getEnvOrThrow('EAM_AUTH_USER'),
+        INFOR_PASSWORD: getEnvOrThrow('EAM_AUTH_PASSWORD'),
+        INFOR_TENANT: getEnvOrThrow('EAM_AUTH_TENANT'),
+        INFOR_ORGANIZATION: getEnvOrThrow('EAM_AUTH_ORGANIZATION'),
       },
     });
-
-    const data = await response.text();
-
+    const data = await response.json();
```

Comment:
Could you move this line after the response.ok check?

Reply:
@janosbabik in case of any 400 errors, we get a error response as a json and we need to log it. Hence, i need to call this before doing the response.ok check, so that i can send the data into the log.